modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/scim2/rest/api/customSchema/SCIM2CustomSchemaUserTestCase.java

```diff
@@ -145,17 +146,25 @@ private void createClaims() throws Exception {
         loginLogoutClient = new LoginLogoutClient(context);
         cookie = loginLogoutClient.login();
         claimMetadataManagementServiceClient = new ClaimMetadataManagementServiceClient(backendURL, cookie);
-        int claimDialect = claimMetadataManagementServiceClient.getClaimDialects().length;
+        ClaimDialectDTO[] claimDialects = claimMetadataManagementServiceClient.getClaimDialects();
 
-        // Set custom schema dialect.
-        ClaimDialectDTO claimDialectDTO = new ClaimDialectDTO();
-        claimDialectDTO.setClaimDialectURI(CUSTOM_SCHEMA_URI);
-        claimMetadataManagementServiceClient.addClaimDialect(claimDialectDTO);
+        boolean isCustomSchemaDialectExist = Arrays.stream(claimDialects)
+                .anyMatch(claimDialect -> StringUtils.equals(claimDialect.getClaimDialectURI(), CUSTOM_SCHEMA_URI));
 
-        //Set claims
+        if (!isCustomSchemaDialectExist) {
+            // Set custom schema dialect.
+            ClaimDialectDTO claimDialectDTO = new ClaimDialectDTO();
+            claimDialectDTO.setClaimDialectURI(CUSTOM_SCHEMA_URI);
+            claimMetadataManagementServiceClient.addClaimDialect(claimDialectDTO);
+        }
+
+        //Set claims.
         setSimpleAttribute();
         setComplexAttribute();
-        assertEquals(claimMetadataManagementServiceClient.getClaimDialects().length, claimDialect + 1);
+
+        ClaimDialectDTO[] updatedClaimDialects = claimMetadataManagementServiceClient.getClaimDialects();
+        Assert.assertTrue(Arrays.stream(updatedClaimDialects)
+                .anyMatch(dialect -> StringUtils.equals(dialect.getClaimDialectURI(), CUSTOM_SCHEMA_URI)));
```

Comment:
Since the method is about creating claims, shall we add an assertion to check the claims addition as well, apart from validating the dialect?

Reply:
Added